packages/ui/src/primitives/tailwind/Select/index.tsx

```diff
@@ -23,183 +23,271 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
+          labelProps?.position === 'left' && 'flex-row items-center gap-x-2'
+        )}
       >
-        <ul
-          className={twMerge('max-h-40 overflow-auto [&>li]:px-4 [&>li]:py-2', menuClassname)}
-          data-testid="select-input-list"
-        >
-          {filteredOptions.value.map((option, index) => (
-            <li
-              key={index}
-              value={option.value}
-              className={twMerge(
-                'cursor-pointer px-4 py-2 text-theme-secondary',
-                option.disabled ? 'cursor-not-allowed' : 'hover:bg-theme-highlight hover:text-theme-highlight',
-                menuItemClassName
+        {labelProps?.text && (
+          <label className="block text-xs font-medium" ref={labelRef}>
+            <div className="flex flex-row items-center gap-x-1.5">
+              <div className="flex flex-row items-center gap-x-0.5">
+                {required && <span className="text-sm text-[#E11D48]">*</span>}
+                <span className="text-xs text-[#D3D5D9]">{labelProps.text}</span>
+              </div>
+
+              {labelProps?.infoText && (
+                <Tooltip content={labelProps.infoText}>
+                  <HelpIconSm className="text-[#9CA0AA]" />
+                </Tooltip>
               )}
-              data-testid="select-input-list-item"
-              onClick={() => handleOptionItem(option)}
+            </div>
+          </label>
+        )}
+
+        <div
+          ref={ref}
+          className="relative"
+          style={{
+            width: variantToWidth[width]
+          }}
+        >
+          <div
+            onClick={() => {
+              if (!disabled) {
+                setOpen((v) => !v)
+              }
+            }}
+            tabIndex={0}
+            className={twMerge(
+              `relative flex w-full items-center gap-x-2 rounded-md border-[0.5px] border-[#42454D] bg-[#141619] text-[#9CA0AA] ${
+                variantSizes[inputSizeVariant]
+              } ${disabled && 'cursor-not-allowed bg-[#191B1F] text-[#6B6F78]'} transition-colors duration-300`,
+              'focus:outline-none',
+              state === 'success' && 'border-[#10B981]',
+              state === 'error' && 'border-[#C3324B]'
+            )}
+            onKeyUp={(e) => {
+              if (disabled || !open) return
+
+              let newIndex = activeIndex
+
+              if (activeIndex === -1) {
+                if (e.code === 'ArrowUp') {
+                  newIndex = options.length - 1
+                } else if (e.code === 'ArrowDown') {
+                  newIndex = 0
+                }
+              } else if (e.code === 'ArrowUp') {
+                newIndex = (activeIndex - 1 + options.length) % options.length
+              } else if (e.code === 'ArrowDown') {
+                newIndex = (activeIndex + 1) % options.length
+              }
+
+              setActiveIndex(newIndex)
+              if (e.code === 'Enter') {
```

Comment:
lets also include spacebar (user normally uses either spacebar or enter)
```suggestion
              if (['Enter', ' '].includes(e.code)) {
```